### PR TITLE
feat(base): Added `subset` method to replicate `pickset` from Chance.js

### PR DESCRIPTION
**Pull request by richardson-trevor**
This is useful for picking a random subset of elements from an array.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -239,3 +239,17 @@ any.date();
...
+## `subset`
+
+Direct usage of [`pickset()` from chance.js](http://chancejs.com/#pickset).
+Options are passed directly to the chance method so refer to its documentation
+for what is available.
+
+### `subset` example
+
+```js
+any.subset(['one', 'two', 'three', 'four'], 2);
```
Comment (travi) [inline docs/API.md]: i wonder if `any.subset(['one', 'two', 'three', 'four'], {size: 2});` might be more consistent with the patterns used elsewhere in `any`. don't think i have strong feelings about it though. do you have thoughts about that?

**Comment by richardson-trevor** [comment]
You know, I thought about that and was indifferent with their defaulting to
a single element list. Since you’re also thinking that, though, it’s worth
considering. To your point, you could easily argue that it would be more
consistent with the others to add the default size of two. We’d just need
to handle the case where the list is not that size. I’ll have to look, they
may handle that by default anyway since they default to selecting one
element and technically a list could be empty.

**Comment by richardson-trevor** [comment]
This is now updated to default to selecting two. I also verified that chance (actually boils down to array.slice, which is how they do the subset) handles when you have a select quantity that is large than the size of the array.

**Comment by travi** [comment]
the part that i was trying to call out was passing in an options object rather than just a single value. if you're on board with keeping that consistent, lets go that route. also, if you've added a default, could you add that to the docs?

**Comment by richardson-trevor** [comment]
Got this updated, but looks like with recent dependency updates my build now fails. I'm seeing things like the following locally:
```
Transformation error for /Users/trevorr/Development/Home/any/src/base.js ; return original code
Invalid Version: https://registry.npmjs.org/istanbul-lib-instrument/-/istanbul-lib-instrument-1.8.0.tgz
```
but `npm t` still passes. Looks like Travis is having issues pulling deps, though.

**Pull request merged**